File: app/src/main/java/com/gmail/wondergab12/restapi/dependencies/MainViewModelSubGraph.java
package com.gmail.wondergab12.restapi.dependencies;

import androidx.appcompat.app.AppCompatActivity;

import com.gmail.wondergab12.restapi.vm.MainViewModel;

import dagger.BindsInstance;
import dagger.Subcomponent;

@MainViewModelScope
@Subcomponent(modules = MainViewModelModule.class)
public interface MainViewModelSubGraph {

    @Subcomponent.Builder
    interface Builder {

        @BindsInstance
        Builder withActivity(AppCompatActivity activity);

        MainViewModelSubGraph build();
    }

    void inject(MainViewModel viewModel);
}

File: app/src/main/java/com/gmail/wondergab12/restapi/dependencies/MainViewModelSubGraph.kt
package com.gmail.wondergab12.restapi.dependencies

import androidx.appcompat.app.AppCompatActivity
import com.gmail.wondergab12.restapi.vm.MainViewModel
import dagger.BindsInstance
import dagger.Subcomponent

@Subcomponent(modules = [MainViewModelModule::class])
interface MainViewModelSubGraph {

    @Subcomponent.Builder
    interface Builder {

        @BindsInstance
        fun withActivity(activity: AppCompatActivity): Builder

        fun build(): MainViewModelSubGraph
    }

    fun inject(viewModel: MainViewModel)
}
